utils.lua

```diff
@@ -50,74 +83,89 @@ function util:GetRealmSlug(realm)
     return realmSlug or realm
 end
 
-local unitIdentifiers = {"mouseover", "player", "target", "focus", "pet", "vehicle"}
+local unitIdentifiers = {
+    ["mouseover"] = true,
+    ["player"] = true,
+    ["target"] = true,
+    ["focus"] = true,
+    ["pet"] = true,
+    ["vehicle"] = true,
+}
 
 do
     for i=1,40 do
-        table.insert(unitIdentifiers, "raid" .. i)
-        table.insert(unitIdentifiers, "raidpet" .. i)
-        table.insert(unitIdentifiers, "nameplate" .. i)
+        unitIdentifiers["raid" .. i] = true
+        unitIdentifiers["raidpet" .. i] = true
+        unitIdentifiers["nameplate" .. i] = true
     end
 
     for i=1,4 do
-        table.insert(unitIdentifiers, "party" .. i)
-        table.insert(unitIdentifiers, "partypet" .. i)
+        unitIdentifiers["party" .. i] = true
+        unitIdentifiers["partypet" .. i] = true
     end
 
     for i=1,5 do
-        table.insert(unitIdentifiers, "arena" .. i)
-        table.insert(unitIdentifiers, "arenapet" .. i)
+        unitIdentifiers["arena" .. i] = true
+        unitIdentifiers["arenapet" .. i] = true
     end
 
-    for i = 1, MAX_BOSS_FRAMES do
-        table.insert(unitIdentifiers, "boss" .. i)
+    for i=1, MAX_BOSS_FRAMES do
+        unitIdentifiers["boss" .. i] = true
     end
 
-    for k=1,table.getn(unitIdentifiers) do
-        table.insert(unitIdentifiers, k .. "target")
+    for k,_ in pairs(unitIdentifiers) do
+        unitIdentifiers[k .. "target"] = true
     end
 end
 
--- Can be called with isUnit("Marahin"), IsUnit("party1"))
-function util:IsUnit(identifier, isPlayer)
-    if not isPlayer and type(identifier) == "string" and identifier:find("-", nil, true) then
-        isPlayer = true
+-- Can be called with isUnit("raid14"), IsUnit("party1"))
+function util:IsUnit(identifier)
```

Comment:
This function is rewritten as the second argument was basically never used as `isPlayer` and caused a potential bug in other scenarios. Previously it also tried to resolve player names in `UnitExists` and `UnitIsPlayer`, but those only work with unit names, not player names.